web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -177,19 +190,31 @@ const AppLaunch = ({ app }: { app: App }) => {
       </ButtonBorder>
     );
   }
+  function handleSamlAppEditButtonClick() {
+    setResourceSpec({
+      name: name,
+      event: DiscoverEventResource.SamlApplication,
+      kind: ResourceKind.SamlApplication,
+      samlMeta: { preset: samlAppPreset },
+      icon: 'Application',
+      keywords: 'saml',
+    });
+  }
   if (samlApp) {
     return (
-      <ButtonBorder
-        as="a"
+      <ButtonWithMenu
```

Comment:
Shouldn't we show `ButtonBorder` when `setResourceSpec` is empty?
Now when I run OSS, I see an edit button on a SAML app that does nothing.

Reply:
Now returns `ButtonBorder` if setResourceSpec is empty https://github.com/gravitational/teleport/pull/43871/commits/18a5608fafa9e8423574eaf1537ad39905a9f6fc